lib/ledger_sync/adaptors/quickbooks_online/expense/operations/update.rb

```diff
@@ -0,0 +1,99 @@
+module LedgerSync
+  module Adaptors
+    module QuickBooksOnline
+      module Expense
+        module Operations
+          class Update < Operation::Update
+            class Contract < LedgerSync::Adaptors::Contract
+              schema do
+                required(:ledger_id).filled(:string)
+                required(:account).hash(Types::Reference)
+                required(:vendor).hash(Types::Reference)
+                required(:amount).filled(:integer)
+                required(:currency).filled(:string)
+                required(:memo).filled(:string)
+                required(:payment_type).filled(:string)
+                required(:transaction_date).filled(:string)
+                required(:line_items).array(Types::Reference)
+              end
+            end
+
+            private
+
+            def build
+              build_account_operation(resource.account)
+              resource.line_items.each do |line_item|
+                build_account_operation(line_item.account)
+              end
+              build_vendor_operation
+              add_root_operation(self)
+            end
+
+            def operate
+              ledger_resource_data = adaptor.find(
+                resource: 'purchase',
+                id: resource.ledger_id
+              )
+              response = adaptor.upsert(
+                resource: 'purchase',
+                payload: merge_into(from: local_resource_data, to: ledger_resource_data)
+              )
+
+              resource.ledger_id = response.dig('Id')
```

Comment:
Maybe I'm missing this, but should we also not be setting any values returned on the resource as well?  There is a chance they differ or were not set originally on the resource.

Reply:
Yes and no. I woudn't say they will differ, but API will return attributes that we don't use with their default values.

In case of related resources, all we care is `Id` for reference. But to do this properly, we really need some serialize/deserialize methods.